src/funnels/SwapDepositor.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+import { Swapper } from "src/funnels/Swapper.sol";
+import { Depositor } from "src/funnels/Depositor.sol";
+
+interface RolesLike {
+    function canCall(bytes32, address, address, bytes4) external view returns (bool);
+}
+
+interface UniV3PoolLike {
+    function slot0() external view returns (
+        uint160 sqrtPriceX96,
+        int24 tick,
+        uint16 observationIndex,
+        uint16 observationCardinality,
+        uint16 observationCardinalityNext,
+        uint8 feeProtocol,
+        bool unlocked
+    );
+}
+
+contract SwapDepositor {
+    mapping (address => uint256) public wards;
+    mapping (address => uint256) public gaps;     // gaps[gem0] is the maximum acceptable difference between the values of the tokens added as liquidity (calculated in terms of gem0), over which the excess token is sold prior to adding liquidity
+
```

Comment:
There is no real sense of "lot" and "counts" here, so might be hard to pre-configure and let it runs with keepers (also the keepers themselves seem to not be supported yet).
I think whatever automation sample we end up using should be as similar as possible structure-wise to the stable swapper.

Reply:
Resolving as we'll do an automated depositor only.